plugin/middleware.go

```diff
@@ -53,12 +53,14 @@ type SpecGetter func(string) *MiddlewareSpec
 
 // Registry contains currently registered middlewares and used to support pluggable middlewares across all modules of the vulcand
 type Registry struct {
-	specs []*MiddlewareSpec
+	specs              []*MiddlewareSpec
+	notFoundMiddleware Middleware
 }
 
 func NewRegistry() *Registry {
 	return &Registry{
-		specs: []*MiddlewareSpec{},
+		specs:              []*MiddlewareSpec{},
+		notFoundMiddleware: nil,
```

Comment:
I think explicit initialization is redundant in this case, as `nil` is implied by default:

http://play.golang.org/p/erfsuZ1-Y8